froide/foirequest/admin.py

```diff
@@ -1021,8 +1021,57 @@ class DeferredMessageAdmin(admin.ModelAdmin):
         "redeliver_subject",
         "close_request",
     ]
+    readonly_fields = (
+        "failed_authenticity",
+        "message_body",
+        "decoded_mail",
+    )
+    fieldsets = (
+        (
+            None,
+            {
+                "fields": (
+                    "recipient",
+                    "sender",
+                    "request",
+                    "spam",
+                    "delivered",
+                )
+            },
+        ),
+        (
+            _("Message Content"),
+            {
+                "fields": (
+                    "failed_authenticity",
+                    "message_body",
+                )
+            },
+        ),
+        (
+            _("Advanced Options"),
+            {"classes": ["collapse"], "fields": ("mail", "decoded_mail")},
+        ),
+    )
 
-    save_on_top = True
+    def decoded_mail(self, obj):
+        return format_html(
+            "<pre>{}</pre>",
+            obj.decoded_mail(),
+        )
+
+    def failed_authenticity(self, obj):
+        with closing(BytesIO(base64.b64decode(obj.mail))) as stream:
```

Comment:
Nit, but maybe we move email parsing logic into the model and cache the parsed email on the model object? Then we don't need to parse multiple times.